sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1577,10 +1574,28 @@ def create(
         render_kwargs: t.Dict[str, t.Any],
         **kwargs: t.Any,
     ) -> None:
+        is_snapshot_deployable: bool = kwargs["is_snapshot_deployable"]
+        if not is_snapshot_deployable and is_table_deployable:
+            # If the snapshot is not deployable, the query may contain references to non-deployable tables or views.
+            # Therefore, we postpone the creation of the deployable view until the snapshot is deployed to production.
+            logger.info(
+                "Skipping creation of the deployable view '%s' for the non-deployable snapshot",
+                table_name,
+            )
+            return
+
+        if self.adapter.table_exists(table_name):
```

Comment:
won't this break the previous fix back in november where we always force recreation of views?

Reply:
The only reason we forced recreation before is because we created a view with wrong (dev) references in the first place. Now when we defer the creation of the deployable view it's no longer a problem.

Reply:
If I'm not mistaken with have an integration test for this.

Reply:
Actually it doesn't look like we do. I'll make sure to add one.